feat: add `stats/base/ndarray/meankbn`

Bureir opened the pull request:
Resolves #8095

## Description

> What is the purpose of this pull request?

This pull request:

- adds `@stdlib/stats/base/ndarray/meankbn`, an ndarray wrapper around `@stdlib/stats/strided/meankbn` implementing the improved Kahan–Babuška compensation algorithm;
- mirrors the API and behavior of `@stdlib/stats/base/ndarray/mean` (supports empty and singleton inputs, non-unit and negative strides, and non-zero offsets);
- includes comprehensive unit tests adapted from `ndarray/mean`, plus a numerics test demonstrating improved conditioning on `[1e16, 1, -1e16]`;
- adds an example and a benchmark;
- adds README and docs (types, REPL transcript, and equation SVG).

## Related Issues

> Does this pull request have any related issues?

This pull request:

- resolves #8095

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

- [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Bureir committed:
feat(stats/base/ndarray): add meankbn wrapper (improved Kahan–Babuska) with docs, examples, tests, benchmark

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/package.json:
This package.json file is invalid per project conventions.

Bureir replied:
Resolved, PTAL.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/index.js:
This file does not follow project conventions.

Bureir replied:
Resolved, PTAL.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/examples/index.js:
This file does not follow project conventions.

Bureir replied:
Resolved, PTAL.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md:
This README does not follow project conventions. See other packages.

Bureir replied:
Resolved, PTAL.

Bureir committed:
docs(ndarray/meankbn): rewrite README to match project conventions

Bureir committed:
docs/chore(ndarray/meankbn): align README, index.js, and example with project conventions

Bureir committed:
chore(ndarray/meankbn): normalize package.json and address review feedback

Bureir commented:
Addressed review feedback: updated README, aligned index.js and example with project conventions, and normalized package.json. Tests and example pass locally. PTAL. @kgryte

Bureir committed:
docs(ndarray/meankbn): fix module header & JSDoc example to match conventions

Bureir reviewed (commented):
Thanks for the quick review! I’ve aligned the new package with project conventions by:

- Rewriting the README to match the structure of sibling packages (Usage / Notes / Examples / See Also), fixed code fences and wording, and clarified the ndarray call shape.

- Updating index.js and lib/index.js to include license headers, module JSDoc, and proper re-export of ./lib.

- Making the example build a typed ndarray and print the result (mirrors ndarray/mean style).

- Normalizing package.json fields (name, description, license, main, keywords).

- Verifying tests/examples/benchmarks pass locally for this package.

If there are any remaining convention nits (section order, phrasing, example format, or types), I’m happy to adjust. PTAL, thanks! @kgryte

Bureir commented:
@kgryte

headlessNode commented:
@Bureir The `README.md` file needs to updated. For reference [see](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/stats/base/sdsnanmean/README.md?plain=1)

Bureir committed:
Update README.md to fit project conventions

Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir commented:
@headlessNode @kgryte Resolved README.md not following project conventions :) PTAL

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md:
```diff
@@ -0,0 +1,137 @@
+
+-->
+
+# @stdlib/stats/base/ndarray/meankbn
```
```suggestion
# meankbn
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md:
```diff
@@ -0,0 +1,137 @@
+var ndarray = require( '@stdlib/ndarray/base/ctor' );
+var Float64Array = require( '@stdlib/array/float64' );
+var meankbn = require( '@stdlib/stats/base/ndarray/meankbn' );
+
+var buf = new Float64Array( [ 1.0, 3.0, 4.0, 2.0 ] );
+var x = new ndarray( 'float64', buf, [ buf.length ], [ 1 ], 0, 'row-major' );
+
+// ndarray variants accept an array-like of ndarrays:
+var v = meankbn( [ x ] );
+// returns 2.5
```
```suggestion
var meankbn = require(  '@stdlib/stats/base/ndarray/meankbn'  );
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md:
```diff
@@ -0,0 +1,137 @@
+
+- If provided an empty ndarray, the function returns `NaN`.
+- The wrapper supports non-unit and negative strides and non-zero offsets.
+- This package is a thin ndarray wrapper around [`@stdlib/stats/strided/meankbn`][@stdlib/stats/strided/meankbn]. Accuracy benefits come from compensated summation in the strided core.
```
```suggestion
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md:
```diff
@@ -0,0 +1,137 @@
+var buf = new Float64Array( [ 1.0, 3.0, 4.0, 2.0 ] );
+var x = new ndarray( 'float64', buf, [ buf.length ], [ 1 ], 0, 'row-major' );
+
+console.log( meankbn( [ x ] ) );
```
```suggestion
var discreteUniform = require( '@stdlib/random/array/discrete-uniform' );
var ndarray = require( '@stdlib/ndarray/base/ctor' );
var ndarray2array = require( '@stdlib/ndarray/to-array' );
var meankbn = require( '@stdlib/stats/base/ndarray/meankbn' );

var xbuf = discreteUniform( 10, -50, 50, {
    'dtype': 'float64'
});
var x = new ndarray( 'float64', xbuf, [ xbuf.length ], [ 1 ], 0, 'row-major' );
console.log( ndarray2array( x ) );

var v = meankbn( [ x ] );
console.log( v );
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md:
```diff
@@ -0,0 +1,137 @@
+
+The function has the following parameters:
+
+- **arrays**: array-like object containing a single input ndarray.
+
+**Returns**: arithmetic mean as a `number`.
+
+</section>
```
```suggestion
The function has the following parameters:

- **arrays**: array-like object containing a single input ndarray.

</section>
```

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md:
```diff
@@ -0,0 +1,127 @@
+<!-- </equation> -->
+
+This package provides an ndarray wrapper which maps a one-dimensional ndarray's buffer, length, stride, and offset into the corresponding strided kernel in [`@stdlib/stats/strided/meankbn`][@stdlib/stats/strided/meankbn].
+
```
```suggestion

```

Bureir replied:
Understood! :) Thank you!

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md:
```diff
@@ -0,0 +1,127 @@
+#### meankbn( arrays )
+
+Computes the [arithmetic mean][arithmetic-mean] of a one-dimensional ndarray using the improved Kahan-Babuska algorithm.
+
```
````suggestion

```javascript
var ndarray = require( '@stdlib/ndarray/base/ctor' );

var xbuf = [ 1.0, 3.0, 4.0, 2.0 ];
var x = new ndarray( 'generic', xbuf, [ 4 ], [ 1 ], 0, 'row-major' );

var v = meankbn( [ x ] );
// returns 2.5
```

````

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md:
```diff
@@ -0,0 +1,127 @@
+* * *
+
+## See Also
+
+- <span class="package-name">[`@stdlib/stats/base/ndarray/mean`][@stdlib/stats/base/ndarray/mean]</span><span class="delimiter">: </span><span class="description">compute the arithmetic mean of a one-dimensional ndarray.</span>
+- <span class="package-name">[`@stdlib/stats/strided/meankbn`][@stdlib/stats/strided/meankbn]</span><span class="delimiter">: </span><span class="description">compute the arithmetic mean using the improved Kahan-Babuska algorithm for strided arrays.</span>
+
+</section>
+
+<!-- /.related -->
+
+<section class="links">
+
+[arithmetic-mean]: https://en.wikipedia.org/wiki/Arithmetic_mean
+
+[@stdlib/stats/base/ndarray/mean]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/ndarray/mean
+
+[@stdlib/stats/strided/meankbn]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/strided/meankbn
+
+</section>
+
+<!-- /.links -->
```
```suggestion
<!-- /.examples -->

<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->

<section class="related">

</section>

<!-- /.related -->

<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="links">

[arithmetic-mean]: https://en.wikipedia.org/wiki/Arithmetic_mean

</section>

<!-- /.links -->
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/package.json:
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "@stdlib/stats/base/ndarray/meankbn",
+  "version": "0.0.0",
+  "description": "Compute the arithmetic mean of a one-dimensional ndarray using the improved Kahan–Babuška algorithm.",
+  "license": "Apache-2.0",
+  "main": "./index.js",
+  "keywords": [
+    "stdlib",
+    "statistics",
+    "stats",
+    "mean",
+    "average",
+    "ndarray",
+    "kahan",
+    "babuska",
+    "compensated",
+    "floating-point",
+    "numerics"
+  ]
+}
```
```suggestion
{
  "name": "@stdlib/stats/base/ndarray/meankbn",
  "version": "0.0.0",
  "description": "Compute the arithmetic mean of a one-dimensional ndarray using the improved Kahan-Babuska algorithm.",
  "license": "Apache-2.0",
  "author": {
    "name": "The Stdlib Authors",
    "url": "https://github.com/stdlib-js/stdlib/graphs/contributors"
  },
  "contributors": [
    {
      "name": "The Stdlib Authors",
      "url": "https://github.com/stdlib-js/stdlib/graphs/contributors"
    }
  ],
  "main": "./lib",
  "directories": {
    "benchmark": "./benchmark",
    "doc": "./docs",
    "example": "./examples",
    "lib": "./lib",
    "test": "./test"
  },
  "types": "./docs/types",
  "scripts": {},
  "homepage": "https://github.com/stdlib-js/stdlib",
  "repository": {
    "type": "git",
    "url": "git://github.com/stdlib-js/stdlib.git"
  },
  "bugs": {
    "url": "https://github.com/stdlib-js/stdlib/issues"
  },
  "dependencies": {},
  "devDependencies": {},
  "engines": {
    "node": ">=0.10.0",
    "npm": ">2.7.0"
  },
  "os": [
    "aix",
    "darwin",
    "freebsd",
    "linux",
    "macos",
    "openbsd",
    "sunos",
    "win32",
    "windows"
  ],
  "keywords": [
    "stdlib",
    "stdmath",
    "statistics",
    "stats",
    "mathematics",
    "math",
    "average",
    "avg",
    "mean",
    "arithmetic mean",
    "central tendency",
    "ndarray"
  ],
  "__stdlib__": {}
}
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/benchmark/benchmark.js:
```diff
@@ -0,0 +1,102 @@
+var pow = require( '@stdlib/math/base/special/pow' );
+var ndarray = require( '@stdlib/ndarray/base/ctor' );
+var pkg = require( './../package.json' ).name;
+var mean = require( './../lib' );
```
```suggestion
var meankbn = require( './../lib' );
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/benchmark/benchmark.js:
```diff
@@ -0,0 +1,102 @@
+
+		b.tic();
+		for ( i = 0; i < b.iterations; i++ ) {
+			v = mean( [ x ] );
```
```suggestion
			v = meankbn( [ x ] );
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/repl.txt:
```diff
@@ -0,0 +1,31 @@
+
+{{alias}}( arrays )
+    Computes the arithmetic mean of a one-dimensional ndarray.
```
```suggestion
    Computes the arithmetic mean of a one-dimensional ndarray using the improved
    Kahan-babuska algorithm.
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/examples/index.js:
```diff
@@ -0,0 +1,11 @@
+var buf = new Float64Array( [ 1.0, 3.0, 4.0, 2.0 ] );
+var x = new ndarray( 'float64', buf, [ buf.length ], [ 1 ], 0, 'row-major' );
+
+console.log( 'mean: %d', meankbn( [ x ] ) );
```
```suggestion
/**
* @license Apache-2.0
*
* Copyright (c) 2025 The Stdlib Authors.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

'use strict';

var discreteUniform = require( '@stdlib/random/array/discrete-uniform' );
var ndarray = require( '@stdlib/ndarray/base/ctor' );
var ndarray2array = require( '@stdlib/ndarray/to-array' );
var meankbn = require( './../lib' );

var xbuf = discreteUniform( 10, -50, 50, {
	'dtype': 'generic'
});
var x = new ndarray( 'generic', xbuf, [ xbuf.length ], [ 1 ], 0, 'row-major' );
console.log( ndarray2array( x ) );

var v = meankbn( [ x ] );
console.log( v );

```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/lib/index.js:
```diff
@@ -0,0 +1,40 @@
+* var meankbn = require( '@stdlib/stats/base/ndarray/meankbn' );
+*
+* var buf = new Float64Array( [ 1.0, 3.0, 4.0, 2.0 ] );
+* var x = new ndarray( 'float64', buf, [ buf.length ], [ 1 ], 0, 'row-major' );
+*
+* var v = meankbn( [ x ] );
+* // returns 2.5
+*/
+
+var main = require( './main.js' );
+
+module.exports = main;
```
```suggestion

// MODULES //

var main = require( './main.js' );


// EXPORTS //

module.exports = main;
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/types/index.d.ts:
```diff
@@ -0,0 +1,45 @@
+import { ndarray } from '@stdlib/types/ndarray';
+
+/**
+* Computes the arithmetic mean of a one-dimensional ndarray.
```
```suggestion
* Computes the arithmetic mean of a one-dimensional ndarray using the improved Kahan-Babuska algorithm.
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/types/index.d.ts:
```diff
@@ -0,0 +1,45 @@
+* var v = mean( [ x ] );
+* // returns 2.5
+*/
+declare function mean<T extends ndarray = ndarray>( arrays: [ T ] ): number;
```
```suggestion
declare function meankbn<T extends ndarray = ndarray>( arrays: [ T ] ): number;
```

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/types/index.d.ts:
```diff
@@ -0,0 +1,45 @@
+
+// EXPORTS //
+
+export = mean;
```
```suggestion
export = meankbn;
```

Bureir committed:
ci: retrigger tests for PR #8116

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/types/test.ts:
```diff
@@ -0,0 +1,57 @@
+/* eslint-disable space-in-parens */
+
+import zeros = require( '@stdlib/ndarray/zeros' );
+import mean = require( './index' );
```
```suggestion
import meankbn = require( './index' );
```

Name needs changing throughout this file

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/test/test.js:
```diff
@@ -0,0 +1,172 @@
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var isPositiveZero = require( '@stdlib/math/base/assert/is-positive-zero' );
+var ndarray = require( '@stdlib/ndarray/base/ctor' );
+var mean = require( './../lib' );
```
Name needs changing throughout this file

Bureir replied:
@headlessNode From mean to meankbn?

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/test/test_numerics.js:
```diff
@@ -0,0 +1,31 @@
+'use strict';
```
This file needs to be deleted.

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/index.js:
```diff
@@ -0,0 +1,39 @@
+/**
```
This file needs to be deleted

headlessNode reviewed (changes requested):
@Bureir The main thing here is to be consistent throughout the PR and following the project conventions.

Bureir commented:
@headlessNode Thank you! I had corrected it as you proposed however there was user error which caused minimal changes so I do apologize.

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/README.md

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/package.json

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/benchmark/benchmark.js

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/types/test.ts

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/types/index.d.ts

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/benchmark/benchmark.js

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/repl.txt

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/examples/index.js

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/lib/index.js

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/types/index.d.ts

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/types/index.d.ts

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Delete lib/node_modules/@stdlib/stats/base/ndarray/meankbn/index.js

Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Delete lib/node_modules/@stdlib/stats/base/ndarray/meankbn/test/test_numerics.js

Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
changes mean to meankbn to follow project conventions

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/types/test.ts:
```diff
@@ -0,0 +1,57 @@
+		'dtype': 'generic'
+	});
+
+	mean( [ x ] ); // $ExpectType number
```
```suggestion
	meankbn( [ x ] ); // $ExpectType number
```
Applies here and for function calls below in this file

headlessNode commented on lib/node_modules/@stdlib/stats/base/ndarray/meankbn/test/test.js:
```diff
@@ -0,0 +1,172 @@
+	t.end();
+});
+
+tape( 'the function calculates the arithmetic meankbn of a one-dimensional ndarray', function test( t ) {
```
```suggestion
tape( 'the function calculates the arithmetic mean of a one-dimensional ndarray', function test( t ) {
```

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/test/test.js

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
Update lib/node_modules/@stdlib/stats/base/ndarray/meankbn/docs/types/test.ts

Co-authored-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>
Signed-off-by: Bureir AL <127953808+Bureir@users.noreply.github.com>

Bureir committed:
ci: retrigger tests for PR #8116

Bureir reviewed (commented):
Reviewing, one moment

Bureir committed:
fixed naming to match convention in main.js

Bureir committed:
Went through and validated every file to meet project conventions by visual comparison against ndarray mean and strided meankbn

Bureir committed:
Went through and validated every file to meet project conventions by visual comparison against ndarray mean and strided meankbn

Bureir committed:
Updated to meet project conventions

Bureir commented:
@headlessNode Fully reviewed and fixed, there was a local version issue and this should almost perfectly meet project conventions unless I misunderstood something. Please see my commit title, I went through each file. I could have missed something but can fix things quickly.

headlessNode reviewed (commented):
LGTM. @kgryte will do the final check & merge.

Bureir reviewed (commented):
All verified, approved by @headlessNode waiting on @kgryte

Bureir committed:
fixed spacing

Bureir commented:
@kgryte @headlessNode Fixed spacing

kgryte committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM after clean-up.

kgryte commented:
Thank you for your contribution and congratulations on your first PR!
